code/test-utils/src/test/java/com/adobe/marketing/tester/util/JSONAssertsParameterizedTests.kt

```diff
@@ -0,0 +1,416 @@
+                    Array(2) {"""{ "k.": 1 }"""}, // Dot at the end of key
+                    Array(2) {"""{ "\"": 1 }"""}, // Escaped double quote
+                    Array(2) {"""{ "'": 1 }"""}, // Single quote
+                    Array(2) {"""{ "\'": 1 }"""}, // Escaped single quote
+                    Array(2) {"""{ "key with space": 1 }"""}, // Space in key
+                    Array(2) {"""{ "\n": 1 }"""}, // Control character
+                    Array(2) {"""{ "key \t \n newline": 1 }"""}, // Control characters in key
+                    Array(2) {"""{ "안녕하세요": 1 }"""}, // Non-Latin characters
+                )
+            }
+        }
+
+        @Test
+        fun `should match special key JSONs`() {
+            assertEqual(expected, actual)
+            assertExactMatch(expected, actual)
+            assertTypeMatch(expected, actual)
+        }
+    }
+
+    @RunWith(Parameterized::class)
+    class AlternatePathValueTest(private val keypath: String, expectedJSONString: String, actualJSONString: String) {
+        private inline fun <T> tryOrNull(block: () -> T): T? {
+            return try {
+                block()
+            } catch (e: Throwable) {
+                null
+            }
+        }
+
+        private val expected: Any = (tryOrNull { JSONObject(expectedJSONString) } ?: tryOrNull { JSONArray(expectedJSONString) })!!
+        private val actual: Any? = tryOrNull { JSONObject(actualJSONString) } ?: tryOrNull { JSONArray(actualJSONString) }
+
+        companion object {
+            private fun createParameterCase(key: String, value: Any?, transform: (Any?) -> String): Array<String> {
+                return arrayOf(key, transform(value), transform(value))
+            }
+
+            private val values = listOf(1, 2.0, "a", true, JSONObject(), JSONArray(), JSONObject.NULL, null)
+
+            @JvmStatic
+            @Parameterized.Parameters(name = "{index}: test with key={0}, expected={1}, actual={2}")
+            fun data(): Collection<Array<String>> {
+                return listOf(
+                    *values.map { value -> createParameterCase("key1", value) { """{ "key1": $it }""" } }.toTypedArray(),
+                    *values.map { value -> createParameterCase("[0]", value) { """[$it]""" } }.toTypedArray(),
+                    *values.map { value -> createParameterCase("[*]", value) { """[$it]""" } }.toTypedArray(),
+                )
+            }
+        }
+
+        @Test
+        fun `should not fail because of alternate path`() {
+            assertExactMatch(expected, actual, typeMatchPaths = listOf(keypath))
+            assertTypeMatch(expected, actual, exactMatchPaths = listOf(keypath))
+        }
+    }
+
+    @RunWith(Parameterized::class)
+    class AlternatePathTypeTest(private val keypath: String, expectedJSONString: String, actualJSONString: String) {
+        private inline fun <T> tryOrNull(block: () -> T): T? {
```

Comment:
same comment for the alternatePath tests, expanding then allows the developer to see the exact expected/actual combinations tested

Reply:
sounds good, lets use the same format for testCase for AlternatePathValueTest